hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictorImpl.java

```diff
@@ -34,28 +34,31 @@
  * Evictor helper methods.
  */
 public class EvictorImpl implements Evictor {
-
     protected final EvictionChecker evictionChecker;
     protected final IPartitionService partitionService;
     protected final MapEvictionPolicy mapEvictionPolicy;
 
+    private final int batchSize;
+
     public EvictorImpl(MapEvictionPolicy mapEvictionPolicy,
-                       EvictionChecker evictionChecker, IPartitionService partitionService) {
+                       EvictionChecker evictionChecker, IPartitionService partitionService, int batchSize) {
         this.evictionChecker = checkNotNull(evictionChecker);
         this.partitionService = checkNotNull(partitionService);
         this.mapEvictionPolicy = checkNotNull(mapEvictionPolicy);
+        this.batchSize = batchSize;
     }
 
     @Override
     public void evict(RecordStore recordStore, Data excludedKey) {
         assertRunningOnPartitionThread();
 
-        EntryView evictableEntry = selectEvictableEntry(recordStore, excludedKey);
-        if (evictableEntry == null) {
-            return;
+        for (int i = 0; i < batchSize; i++) {
```

Comment:
further improvement can be to continue this loop only if map needs eviction according to evictionChecker, by doing this, we can prevent unneeded removal of entries.

Reply:
I was thinking about it - it was actually the original implementation. but it adds overhead - because of the extra size checking.

Reply:
For size based eviction i don't think it can cause much cost but for heap based you can be right. 

Other thing that can be costly is `selectEvictableEntry`, i remember we saw that cost in a bulk put operation before. Maybe we can think to remove more than 1 entry from the set of sampled ones.

Reply:
yes, that's certainly a possible improvement in the future. 

Increasing the sample size `n`-times and then picking `n` candidates will increase accuracy of the probabilistic algorithm.

Reply:
also maybe worth to note that this setting will also be affecting HD eviction.